### PR TITLE
add gops dependency and utilize it in Cilium agent

### PR DESCRIPTION
**Pull request by ianvernon**
Add support for getting stack traces of the Cilium agent with gops, a profiling tool from Google.

Partially address #1637 

Signed-off by: Ian Vernon <ian@cilium.io>

### CODE REVIEW

File: daemon/main.go [hunk 1]
```diff
@@ -108,6 +109,12 @@ var RootCmd = &cobra.Command{
 }
 
 func main() {
+
+	// Open socket for using gops to get stacktraces of the agent.
+	if err := gops.Listen(gops.Options{}); err != nil {
```
Comment (aanm) [inline daemon/main.go:116]: Add this only in debug.

**Review by aanm — DISMISSED** [review]

File: daemon/main.go (hunk 1 above)
Comment (rlenglet) [reply daemon/main.go:116]: I disagree. This is a production troubleshooting tool, not a debug tool. This must be always on, in both development and production. It has zero overhead, so I don't see any reason not to do this.

**Review by rlenglet — APPROVED** [review]

**Comment by rlenglet** [comment]
This doesn't build. You seem to have forgotten to vendor gops.

File: daemon/main.go (hunk 1 above)
Comment (ianvernon) [reply daemon/main.go:116]: I'm with Romain on this - we need this precisely when we hit an issue and need diagnostics.

File: daemon/main.go [hunk 2]
```diff
@@ -110,6 +111,12 @@ var RootCmd = &cobra.Command{
...
+		log.Fatal(err)
```
Comment (tgraf) [inline daemon/main.go]: Can you change this to `log.WithError(err).Fatal("Unable to initialize gops")`? I assume the error returned by  `Listen()` does not include that this is about gops?

Comment (tgraf) [reply daemon/main.go]: Did you test that logging is set up properly at this point?

**Review by tgraf — CHANGES_REQUESTED** [review]

File: daemon/main.go (hunk 2 above)
Comment (ianvernon) [reply daemon/main.go]: It logs to `stderr`, but I will have the error case mimic the behavior of if the execution of the following command fails:
```
... err != nil {
  fmt.Println(err)
  os.Exit(-1)
}
```

File: daemon/main.go [hunk 3]
```diff
@@ -110,6 +111,13 @@ var RootCmd = &cobra.Command{
 }
 
 func main() {
+
+	// Open socket for using gops to get stacktraces of the agent.
+	if err := gops.Listen(gops.Options{}); err != nil {
+		fmt.Println(err)
```
Comment (tgraf) [inline daemon/main.go]: What error message will this print? Will it be clear that this is gops related?

**Review by tgraf — DISMISSED** [review]

File: daemon/main.go (hunk 3 above)
Comment (ianvernon) [reply daemon/main.go]: I will add a custom error message like the following:
`errorString := fmt.Sprintf("unable to start gops: %s", err)
fmt.Println(errorString)`

Comment (ianvernon) [reply daemon/main.go]: Fixed.

**Review by tgraf — APPROVED** [review]

**Pull request merged**